package NYT.NNodeTrackerClient.NProto;

import "yt/core/misc/guid.proto";
import "yt/core/misc/error.proto";
import "yt/ytlib/node_tracker_client/node.proto";
import "yt/ytlib/chunk_client/chunk.proto";

////////////////////////////////////////////////////////////////////////////////

// Chunk Manager stuff.

message TChunkAddInfo
{
    required NYT.NProto.TGuid chunk_id = 1;
    required bool cached = 2;
    required NYT.NChunkClient.NProto.TChunkInfo chunk_info = 3;
}

message TChunkRemoveInfo
{
    required NYT.NProto.TGuid chunk_id = 1;
    required bool cached = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReqRegisterNode
{
    required TNodeDescriptor node_descriptor = 1;
    required TNodeStatistics statistics = 2;
    required NYT.NProto.TGuid cell_guid = 3;
}

message TRspRegisterNode
{
    required int32 node_id = 1;
    required NYT.NProto.TGuid cell_guid = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReqFullHeartbeat
{
    required int32 node_id = 1;
    required TNodeStatistics statistics = 2;

    // Chunk Manager stuff.
    repeated TChunkAddInfo chunks = 3;
}

message TRspFullHeartbeat
{ }

////////////////////////////////////////////////////////////////////////////////

message TReqIncrementalHeartbeat
{
    required int32 node_id = 1;

    // Node Tracker stuff.
    required TNodeStatistics statistics = 2;
    repeated string alerts = 5;

    // Chunk Manager stuff.
    repeated TChunkAddInfo added_chunks = 3;
    repeated TChunkRemoveInfo removed_chunks = 4;
}

message TRspIncrementalHeartbeat
{ }

////////////////////////////////////////////////////////////////////////////////
